File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num: nums) {
            set.add(num);
        }

        int res = 0;

        for (int num: nums) {
            int streak = 0;
            while (set.contains(num)) {
                streak++;
                num++;
            }
            res = Math.max(res, streak);
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    lengthOfLIS(nums) {
        const tail = []

        for (const num of nums) {
            let l=0, r= tail.length
            while (l<r) {
                const mid = Math.floor((r+l)/2)
                if (tail[mid] < num) {
                    l = mid + 1
                } else {
                    r = mid
                }
            }
            tail[l] = num
        }

        return tail.length
    }
}
